### PR TITLE
fix: making the computation of `next_bp_hash` consistent with the lig…

### PR DESCRIPTION
**Pull request by SkidanovAlex**
…htclient spec

Test plan
---------
Adding the verification to the `lightclnt` python test.
Besides not verifying the hash, it also wasn't verifying the approval signatures, adding it as well.

### CODE REVIEW

File: chain/chain/src/chain.rs [hunk 1]
```diff
@@ -383,20 +382,9 @@ impl Chain {
 
     pub fn compute_bp_hash_inner<'a, T>(bps: T) -> Result<CryptoHash, Error>
```
Comment (bowenwang1996) [inline chain/chain/src/chain.rs:383]: It seems that you don't really need this function given the way it is right now

Comment (SkidanovAlex) [reply chain/chain/src/chain.rs:383]: It's still called from three different places (test, block production and genesis)

**Comment by nearmax** [comment]
Adding automerge since it is blocking @k06a

**Comment by SkidanovAlex** [comment]
It doesn't compile yet, I'm fixing it locally, so automerge won't help much :)

**Comment by nearmax** [comment]
It is currently a blocker for @k06a .

**Comment by bowenwang1996** [comment]
@SkidanovAlex also as I mentioned you actually changed the protocol here so it is very desirable to bump protocol version.

**Pull request merged**